code/addons/test/src/postinstall.ts

```diff
@@ -53,6 +55,34 @@ export default async function postInstall(options: PostinstallOptions) {
   // if Vitest is installed, we use the same version to keep consistency across Vitest packages
   const vitestVersionToInstall = vitestVersionSpecifier ?? 'latest';
 
+  if (info.frameworkPackageName === '@storybook/nextjs') {
+    const out = await prompts({
+      type: 'confirm',
+      name: 'migrateToExperimentalNextjsVite',
+      message: `To use this addon, you should migrate to use @storybook/experimental-nextjs-vite, do you want to migrate?`,
+      initial: true,
+    });
+
+    if (out.migrateToExperimentalNextjsVite) {
+      await packageManager.addDependencies({ installAsDevDependencies: true }, [
+        '@storybook/experimental-nextjs-vite',
+      ]);
+
+      await packageManager.removeDependencies({}, ['@storybook/next']);
+
+      // update the storybook main.ts config file
+      const mainJsPath = serverResolve(resolve(options.configDir, 'main')) as string;
+
+      const config = await readConfig(mainJsPath);
+
+      config.setFieldValue(['framework'], '@storybook/experimental-nextjs-vite');
```

Comment:
If users have a monorepo-shenanigan like the it has to be respected as well e.g.:

```ts
const getAbsolutePath = (packageName) =>
  path.dirname(require.resolve(path.join(packageName, 'package.json')));

export default {
  framework: {
    name: getAbsolutePath('@storybook/react'),
    options: {},
  },
}
```

Reply:
got it, I think I have a solution for this.